projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx

```diff
@@ -0,0 +1,89 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Returns a notice box for displaying notices about record count and plan limits
+ *
+ * @param {object} props - record count, record limit, hasbeenindexed, hasvaliddata, hasItems
+ * @returns {React.Component} notice box component.
+ */
+export function NoticeBox( props ) {
+	const notices = [];
+
+	// check data is valid
+	if ( props.hasValidData === false ) {
+		notices.push( {
+			message: __(
+				"We weren't able to properly locate your content for Search",
+				'jetpack-search-pkg'
+			),
+			isImportant: true,
+		} );
+	}
+
+	// check site has been indexed
+	if ( props.hasBeenIndexed === false ) {
+		notices.push( {
+			message: __( 'Your content has not yet been indexed for Search', 'jetpack-search-pkg' ),
+		} );
+	}
+
+	// check at least one indexable item
+	if ( props.hasItems === false ) {
+		notices.push( {
+			message: __(
+				"We weren't able to locate any content to Search to index. Perhaps you don't yet have any posts or pages?",
+				'jetpack-search-pkg'
+			),
+		} );
+	}
+
+	if ( props.recordCount > props.planRecordLimit ) {
+		notices.push( {
+			message: sprintf(
+				// translators: %1$d: site's current plan record limit, %2$d: record limit of the next plan up
+				__(
+					'You recently surpassed %1$d records and will be automatically upgraded to the next billing tier of %2$d max records',
+					'jetpack-search-pkg'
+				),
+				props.planRecordLimit,
+				props.planRecordLimit * 10 //TODO: this is currently hard coded & incorrect. Needs to have the next tier plan record limit added
```

Comment:
Shall we start accepting this as a prop so we're ready to plug in the real value? `nextTierPlanRecordLimit` or similar?

Reply:
Or alternatively, as we discussed, alter the wording so the next limit is not mentioned. @annacmc said:

"The alternative is to alter the wording to just be 'to the next tier' with a link to the tier/pricing page."

Reply:
have updated it for the simpler wording for now, along with a TODO to get the link in there to the tier/pricing page (this will make more sense to add once the notice box is displaying in a dismissible box form